File: src/Patterns/MVC/NotMVC/GooseAdapter.java
package Patterns.MVC.NotMVC;

/**
 * Created by Home on 11.12.2016.
 */
public class GooseAdapter implements Quackable {
    Goose goose;

    public GooseAdapter(Goose goose) {
        this.goose = goose;
    }

    @Override
    public void quack() {
        goose.honk();
    }

    @Override
    public void registerObserver(Observer observer) {
    }

    @Override
    public void notifyObservers() {

    }
}

File: src/patterns/together/adapter/GooseAdapter.kt
package patterns.together.adapter

import patterns.together.Quackable
import patterns.together.impl.Goose
import patterns.together.observer.Observer
import patterns.together.observer.QuackObservable
import patterns.together.observer.impl.Observable

class GooseAdapter(private val goose: Goose) : Quackable {
    val observable: Observable = Observable(this)

    override fun registerObserver(observer: Observer) {
        observable.registerObserver(observer)
    }

    override fun notifyObservers() {
        observable.notifyObservers()
    }

    override fun quack() {
        goose.honk()
        notifyObservers()
    }
}